app/src/main/java/chat/rocket/android/contacts/ui/ContactsFragment.kt

```diff
@@ -301,9 +301,15 @@ class ContactsFragment : Fragment(), ContactsView {
         return list
     }
 
-    fun queryContacts(query: String) {
+    private fun queryContacts(query: String) {
+        var result: ArrayList<ItemHolder<*>>? = null
         if (query.isBlank() or query.isEmpty()) {
-            setupFrameLayout(contactArrayList)
+            if (hasContactsPermissions()) {
+                setupFrameLayout(contactArrayList)
+            } else {
+                result = arrayListOf()
```

Comment:
is this needed? could this just be setupFrameLayout() like it is elsewhere?

Reply:
Unfortunately it is needed, because we need some flag to create the specific view where we are searching but the search results are empty.  Without the empty array, setupFrameLayout() would render the regular, contacts disabled, view.